From 396f1c399c30d10490e12e731f0c6d59a184620d Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Wed, 2 Aug 2017 14:19:10 +0800
Subject: [PATCH 2275/5242] MLK-16120 gpu: imx: dpu: fetchdecode: Use
 interpolate upsampling mode for NV16/61

commit  4e8e802235932dd08f803f351b8d61c913dda12b from
https://source.codeaurora.org/external/imx/linux-imx.git

According to dpu spec, we should use interpolate upsampling mode for NV16/61.
This patch changes the mode from replicate to interpolate for the two pixel
formats.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-fetchdecode.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/imx/dpu/dpu-fetchdecode.c b/drivers/gpu/imx/dpu/dpu-fetchdecode.c
index 166fa68..c28e110 100644
--- a/drivers/gpu/imx/dpu/dpu-fetchdecode.c
+++ b/drivers/gpu/imx/dpu/dpu-fetchdecode.c
@@ -277,10 +277,12 @@ void fetchdecode_set_fmt(struct dpu_fetchdecode *fd, u32 fmt)
 		is_yuv422upsamplingmode_interpolate = true;
 		need_csc = true;
 		break;
-	case DRM_FORMAT_NV12:
-	case DRM_FORMAT_NV21:
 	case DRM_FORMAT_NV16:
 	case DRM_FORMAT_NV61:
+		is_yuv422upsamplingmode_interpolate = true;
+		/* fall-through */
+	case DRM_FORMAT_NV12:
+	case DRM_FORMAT_NV21:
 		is_planar_yuv = true;
 		is_rastermode_yuv422 = true;
 		is_inputselect_compact = true;
-- 
1.7.9.5

